src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,135 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (head == null) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndexRange(index);
+        Node<T> current = getNodeByIndex(index);
```

Comment:
move checkIndex inside getNodeByIndex

Reply:
I decided that the "checkIndexRange" method would not be useful and moved its implementation to the "getNodeByIndex" method, thanks!